site/docs/spec.md

```diff
@@ -646,6 +646,18 @@ Each partition field in the fields list is stored as an object. See the table fo
 
 In some cases partition specs are stored using only the field list instead of the object format that includes the spec ID, like the deprecated `partition-spec` field in table metadata. The object format should be used unless otherwise noted in this spec.
 
+#### Partition Field ID handling
+
+A partition field id is an integer (starting at 1000) used to identify a partition field.
+
+Since iceberg release 0.8.0, partition fields are present in every partition field of partition specs in a table metadata.
```

Comment:
The spec shouldn't reference Iceberg Java releases. The requirements here are for different versions of tables:

* For v1 tables, partition field metadata should include a `field-id` for each partition field, but this is not required.
* For v2 tables, partition field metadata _must_ include a `field-id` for each partition field.

Then for compatibility within v1, this should recommend not removing fields but replacing them with `always null`. We also need to add that transform to the spec.